backends/ebpf/codeGen.cpp

```diff
@@ -462,6 +462,10 @@ unsigned EBPFInitializerUtils::ebpfTypeWidth(P4::TypeMap *typeMap, const IR::Exp
     auto type = typeMap->getType(expr);
     if (type == nullptr) type = expr->type;
     if (type->is<IR::Type_InfInt>()) return 32;  // let's assume 32 bit for int type
+    if (type->is<IR::Type_Set>()) type = type->to<IR::Type_Set>()->elementType;
+    // TODO: fix this when select expression from parser supports more than one field
```

Comment:
what do you want to fix? Do you mean remove the if statement?

Reply:
Statement supports only the first element in the list, so in case of multiple fields in `select` expression will be generated and used type only for first field